Jk frontend redesign

kulincz opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| We want demoshop to look nice :slightly_smiling_face: 
|New feature| Yes <!-- Do not forget to update CHANGELOG.md and possibly docs/ -->
|BC breaks| No <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

There is video walkthrough
on desktop:
https://www.useloom.com/share/07abd57c6f9740efb153a49a268ee9cf
and mobile:
https://www.useloom.com/share/b38f6149ee3440fbb7b61e49a3a8faf9

kulincz committed:
Menu and header design tests

kulincz committed:
Menu and header design tests

kulincz committed:
Menu and header design tests

kulincz committed:
Menu and header design tests

kulincz committed:
Homepage redesign

kulincz committed:
Styling subcategories

kulincz committed:
Design Category

kulincz committed:
Filters are now in left panel

kulincz committed:
Styling Filters

kulincz committed:
Filters are now responsive

kulincz committed:
Product list redesign

kulincz committed:
Paging is now on the right side at desktop view

kulincz committed:
Filters CSS bugfix on mobile

kulincz committed:
Sorting buttons update

kulincz committed:
Form spinbox has now better UX on tap screens

kulincz committed:
Cart table tunning

kulincz committed:
Product detail and Cart is now in new design

kulincz committed:
Design tunning on buttons

kulincz committed:
Order process is now in new design

kulincz committed:
Logo fixup

kulincz committed:
Pagination is with next / prev icons

kulincz committed:
Text pages and search is in new design

LukasHeinz commented:
Hi @kulincz, nice job!

I have few ideas and recommedations:
1. Category detail: sorting options above product list is not very nice and look too "plaintext". I would recommend to design them as tabs or something like this...
2. Category detail: please remove option to display more bestsellers i.e. there will be only 3 best sellers
3. Cart: progress bar with 3 steps below header could look better...
4. Order>Personal info: please delete block with terms and conditions
5. Homepage & Category detail: please add there some static blocks that will look good. For example:
- on category list next to sucategorie there sould be some "Call us" blosk 
<img width="287" alt="snimek obrazovky 2018-08-09 v 16 24 34" src="https://user-images.githubusercontent.com/33071107/43905243-f2622f78-9bf0-11e8-84ca-680d830a086f.png">
- on homepage ther should be some line with some values
<img width="1207" alt="snimek obrazovky 2018-08-09 v 16 25 52" src="https://user-images.githubusercontent.com/33071107/43905244-f45bcdfc-9bf0-11e8-83ba-832e56be42c4.png">
6. Category detail>Filters: please remove possibility to manually set values for price filter, leve ther only price slider and numbers as values.

The point of redesign is that it need to be looking bit different than stable ;-)

vitek-rostislav committed:
dependency on shopsys/framework upgraded to the recent dev-master version

vitek-rostislav committed:
FE homepage: prepared three new banner positions that are displayed instead of the slider

kulincz committed:
Added Homepage Banners

kulincz committed:
Removed option (button) to show more than 3 TOP products

kulincz committed:
Sorting is now in new design

kulincz committed:
Cart progress bar is now in new design

kulincz committed:
Deleted block with terms and condition

kulincz committed:
Price slider is now without inputs and showing only values

kulincz committed:
Fixed layout bad using

kulincz committed:
Footer is now all black and newly contains shop advantages with icons

kulincz committed:
Footer advantages styling

kulincz commented:
Hi @LukasHeinz 1,2,3,4,6 are done.
On 5) is ready Advantages part above footer and tomorrow I will do Call us block. Than it should be all from my point of view .. let me know if it is ok for you as well :-)

kulincz committed:
Filters fixup

kulincz committed:
Added new Call us block in panel

kulincz commented:
Number 5 is done.

vitek-rostislav committed:
fixed acceptance tests after FE redesign

- LoginAsCustomerCest::testLoginAsCustomer() skipped temporarilly (it needs the info bar that is hidden temporarilly as well for design debugging purposes)
- price filter tests leverage direct javascript execution for filling in the hidden price inputs as making codeception to handle the jQueryUI slider would be overkill

LukasHeinz commented:
Hi @kulincz . Nice work on the my request. Everything looks really good. 

One more idea:
7) Order: In order summary in 2ns and 3rd step of order (the panel on the right) everything is just a set of text. Can you make it nicer? :-)
8) Please send me images for new banner positions for homepage (via email) so I can try it (we do not have ready demo data).

Nevertheless I am sending this design and its implementation to marketing so they might have some other ideas. I will let you know

LukasHeinz commented:
@vitek-rostislav How it comes that new banner positions are not mentioned here - http://rv-frontend-redesign.loki.shopsys.com/admin/image/overview/ ? I thought it is automatical.

kulincz commented:
Hi @LukasHeinz thank you for your feedback. 
ad 7) I was check it and I have no idea hot to make it better. Actually I think that on this place is less more. There is no need to set this box apart.
ad 8) There you have images – put them over admin like this:
![image](https://user-images.githubusercontent.com/29253585/44325344-d86e8880-a458-11e8-9018-b428413f0434.png)
And banners:
![homepage-banner-1](https://user-images.githubusercontent.com/29253585/44325355-e45a4a80-a458-11e8-9e6d-b46ab8572bbf.jpg)
![homepage-banner-2](https://user-images.githubusercontent.com/29253585/44325356-e45a4a80-a458-11e8-8d08-0e02ba78eacc.jpg)
![homepage-banner-3](https://user-images.githubusercontent.com/29253585/44325358-e4f2e100-a458-11e8-8da0-2df0966a120a.jpg)
Now I will work on styling Zasilkovna plugin in order process. Is it ok for you this way?

LukasHeinz commented:
Ad 8) Perfect, thank you.

Ad 7) In my point the summary is not much readable if there is several items - there is just a lot of text... Design on agatinsvet.cz is much better (see picture below)

![snimek obrazovky 2018-08-20 v 10 06 51](https://user-images.githubusercontent.com/33071107/44328072-e1635800-a460-11e8-9e97-357570dfbf5d.png)

LukasHeinz commented:
9) Please change favico (it doesn't metter to what)

kulincz committed:
Pick up place Zasilkovna is now in new design, waiting for JS fixup js-responsive-toggle in modal

kulincz committed:
Cart preview is now more fancy and frontend has new favicon

kulincz committed:
Visibility od Pickup place box fixup

kulincz commented:
Ok :-) @LukasHeinz 7) and 9) should be fixed. Check it please.

kulincz committed:
Table cart preview is now sticky while scrolling in modern browsers

kulincz committed:
Pick up place text styling

LukasHeinz commented:
@kulincz You are awesome ;-) I consider all my request done. 

Now it waits for suggestions from marketing & sales

LukasHeinz commented:
This is also connected to issues #11 and #14 . I would recommend to resolve both issue in this branche. Is it OK? @kulincz @vitek-rostislav

vitek-rostislav commented:
Hi @LukasHeinz, I do agree with you, let's fix the issues here. Will you tell us (or at least give us some hints) what to do in those issues or shall we come up with any solution? :slightly_smiling_face: 

> How it comes that new banner positions are not mentioned here - http://rv-frontend-redesign.loki.shopsys.com/admin/image/overview/ ? I thought it is automatical.

No, it is not automatical. You need to do it manually in `images.yml` config and I absolutely forgot about the functionality when implementing https://github.com/shopsys/shopsys/pull/395. I'll add it there in and also the information into the cookbook :wink:

vitek-rostislav committed:
Zasilkovna: removed tooltips and "more info" section containing information about opening hours of pickup points due to UX problems

- approved by the product owner (Lukas Heinz)
- PickupPlace::$description is still persisted but now it is never used

vitek-rostislav committed:
images.yml: added information about new banner positions

vitek-rostislav committed:
Zasilkovna popup: display info massage when no pickup locations are found

vitek-rostislav committed:
todo.js removed: the functionality refactored into rangeSlider.js and productList.AjaxFilter.js

vitek-rostislav committed:
proper currency symbol is now displayed in FE price filter

vitek-rostislav committed:
Zasilkovna tweaking

- empty boxes on transport selection page and in Zasilkovna popup are now hidden
- removed useless Shopsys.pickUpPlaceSelection.updateSummaryVisibility

boris-brtan commented on app/config/parameters.yml.dist:
```diff
@@ -17,3 +17,5 @@ parameters:
 
   secret: ThisTokenIsNotSoSecretChangeIt
+
+  elasticsearch_hosts: ['elasticsearch:9200']
```
seems that in dev-master version there is also 
`microservice_product_search_url: 'http://microservice-product-search:8000'`
but in docker-compose i can`t see microservice-product-search so it is ok

vitek-rostislav replied:
yes, demoshop is based on pre-microservice `master` :slightly_smiling_face: it will not be updated until someone runs `composer update` :wink:

boris-brtan commented on build-dev.xml:
```diff
@@ -24,10 +24,10 @@
-            logoutput="true"
-            passthru="true"
-            checkreturn="true"
+                executable="${path.composer.executable}"
```
seems like IDE aligned it with its children.
```
<exec
                executable="${path.ecs.executable}"
                logoutput="true"
                passthru="true"
                checkreturn="true"
        >
            <arg value="check" />
            <arg path="${path
```

vitek-rostislav replied:
true, fixed in fixup commit

boris-brtan commented on src/Shopsys/ShopBundle/Resources/views/Front/Layout/layout.html.twig:
```diff
@@ -47,9 +47,9 @@
                         <div class="box-advantages__item__text">
                             <div class="box-advantages__item__title">
-                                Great Value
+                                {{ 'Great Value'|trans }}
```
i can see no entries of these values in translations/*.po

vitek-rostislav replied:
added in new commit (it was necessary to add yet another one commit to fix the export itself)

boris-brtan commented on tests/ShopBundle/Acceptance/acceptance/PageObject/Front/ProductFilterPage.php:
```diff
@@ -23,7 +22,11 @@ public function __construct(StrictWebDriver $strictWebDriver, AcceptanceTester $
+         * The input for minimal price is hidden by design so Codeception is not able to fill it directly.
+         * There is a jQueryUI price slider in the filter form but making Codeception to handle it correctly would be overkill.
+         */
+        $this->tester->executeJS(sprintf('$("#product_filter_form_minimalPrice").val(%d).change()', $price));
```
ok. Maybe we could at least check whether the component that handles the input is visible for User.
`%d` -> `%.2f`

vitek-rostislav replied:
after discussing it personally we agreed that checking the visibility is not necessary :slightly_smiling_face: 

> %d -> %.2f  :heavy_check_mark:

boris-brtan commented on src/Shopsys/ShopBundle/Resources/scripts/frontend/todo.js:
```diff
@@ -21,4 +21,19 @@ $(document).ready(function () {
 
+    // vzato z agáty, je potřeba nějak začlenit do rangeSlider.js .. funkčně to bere hodnoty z inputů, které máme skryté a dává je to do textových elementů, které zobrazuji. Po resetu filtrů by se měli aktualizovat, což se nyní neděje.
+    var showSliderPrices = function ( $slider ) {
+		var $minimalPriceInput = $ ( '#product_filter_form_minimalPrice' );
```
it looks like extra spaces from somehow compliled javascript

vitek-rostislav replied:
yes..but the file was only temporary and is completely removed in another commit :slightly_smiling_face:

boris-brtan reviewed (changes requested):
It looks ok to me, i just left some questions

vitek-rostislav committed:
fixup! dependency on shopsys/framework upgraded to the recent dev-master version

vitek-rostislav committed:
TransportFormTypeExtension: constraint message must not be translated via t() function call

- contraint messages are translated automatically
- removed outcommented line as well

vitek-rostislav committed:
dump translation messages and add missing czech translations

vitek-rostislav committed:
fixup! fixed acceptance tests after FE redesign

vitek-rostislav commented:
Hi @boris-brtan, thx for the review :+1: 
I have answered to all your comments and added some additional commits, please check it out, thank you :slightly_smiling_face:

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/menu.less:
```diff
@@ -6,25 +6,27 @@
         &__link {
             position: relative;
+
+            color: @color-0;
```
For me its better to not use default color variables.. think about connect them with context - @color-link @color-link-alternative etc.. its better when you need to change color of all links or when you want to create new template with different colors

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less:
```diff
@@ -44,11 +44,17 @@ img {
 
+    &__header {
+        box-shadow: 0 0 20px rgba(0,0,0,.1);
+        margin-bottom: 30px;
```
if you want to follow rules from shopsys, you should to split css properties to blocks (important / stylistic)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/layout/header/search.less:
```diff
@@ -97,9 +102,9 @@
         }
 
         &__image {
+            flex: 0 30px;
```
if image is a square, is nice to add 30px to local variable

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/cookies.less:
```diff
@@ -1,46 +1,53 @@
+    right: auto;
+    bottom: 10px;
+    left: 10px;
+    right: 10px;
```
when i see same pixel values, i connect them (if they are related) to local variable .. also they can be related to web padding etc - then i connect them with glabal variable like that @box-cookies-outer-gap: @web-padding;

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/categories.less:
```diff
@@ -1,26 +1,26 @@
 
-    @media @query-sm {
-        margin-left: -@list-categories-gap;
+    .wrap-border;
```
im not fan of theese connections between components.. but i understand that it has pros here but also cons (when somebody add something to wrap-border, he will not know, he change something in list-categories component - especially when its defined on the end of properties.. example - margin in wrap-border will change margin here too)

kulincz replied:
I understand ... but maybe here is desirable to change it on every place because it's fragment of desing .. so for consistency is it ok from my point of view. What do you think?

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/categories.less:
```diff
@@ -32,14 +32,32 @@
+            left: 0;
+            bottom: 0;
+            right: 100%;
+            background-color: fade(@color-0, 5%);
```
what about css linther for css blocks? Have you started to use it?

kulincz replied:
My less file is linted all right I think .. check it:
![image](https://user-images.githubusercontent.com/29253585/44836873-648e6600-ac38-11e8-93ed-164808d65eae.png)
It's based on this list: http://wiki/index.php/Less_css_v_Shopsys_5.6,_BEM_a_pravidla_psan%C3%AD_CSS

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/wrap/border.less:
```diff
@@ -1,3 +1,7 @@
+    border: 3px solid @color-border-light;
+}
+
+.wrap-border--bold {
```
this modification can lead to use it without main component, cos now its something like helper :) i can see problems in far future when somebody esle will add something to main component (idk .. box shadow.. ) and it will not be for example in components when you add only wrap-border--bold

kulincz replied:
Ok, so what do you suggest for solution? For demoshop purpose it's needed to be in these not-using helper guidelines? I can change it if it's necessary.

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -1,22 +1,47 @@
     &__price {
         margin-bottom: @margin-gap;
+
+        .form-line {
```
i  dont like mixing of components.. i propably know why you did it.. for me its nicer to send to theme class connected to this component (box-filter__price__form) what is not so nice eather.. or make new selfstanding component and remove form-line component from here (but it can be harder to do it - maybe you need to change behavior of adding classes to forms in twig theme)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -1,22 +1,47 @@
+            }
+        }
+
+        .ui-slider {
```
could be.. if you dont want to write components to other components and if its not hard, you can wrap ui-slider into new div

kulincz replied:
I don't understand ... why is this problem? I want indent ui-slider just in this case / place ... please show me excatly what is better solution.

seedya replied:
Slider could be wrapped, or can have class like this ".box-filter__slider" just in case js component will change.. so thats not a problem, just thinking in way - component has only element classes and even if you change component inside, you will not need to change ".ui-slider" class here

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/styled-checkbox.less:
```diff
@@ -0,0 +1,65 @@
+	    position: relative;
+
+		font-weight: bold;
+		color: @color-add;
```
indenation

kulincz replied:
Sorry, I don't understand ... in my file is a lot more lines, right indentations :-/ 
![image](https://user-images.githubusercontent.com/29253585/44837375-c8fdf500-ac39-11e8-8193-d5183044af76.png)

kulincz replied:
Path is ok a it is the same file:
![image](https://user-images.githubusercontent.com/29253585/44837429-e3d06980-ac39-11e8-96e1-87e6dc7b8dbf.png)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/styled-checkbox.less:
```diff
@@ -0,0 +1,65 @@
+@in-styled-checkbox-width: 20px;
+
+.in-styled-checkbox {
```
in case it should be style for classic checkbox, why to not use form-choice component? what i do is to use it for all checkboxes, even radio buttons (with modification --radio) can post you example if you want :) but i know, here in filter you need to use special classes cos everytime its something different here, so then i usually use additional box-filter__ classes

seedya replied:
another thing - i think it belongs to FORM components

kulincz replied:
Ok, I renamed it and moved to form component.

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/styled-checkbox.less:
```diff
@@ -0,0 +1,65 @@
+			position: absolute;
+			left: @in-styled-checkbox-width/6;
+			top: @in-styled-checkbox-width/6;
+			content:"\f108";
```
what happen if you add new svg icon to font for example to the start of set? i think f108 will change (not sure) .. for safe use i write :extend(.svg-name:before)

kulincz replied:
Great solution! Kudos for you 👍 .. I didn't know it :)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/styled-checkbox.less:
```diff
@@ -0,0 +1,65 @@
+
+	    	font-family: "svg";
+	    	font-size: 11px;
+			color: white;
```
why you use non hash declaration?

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig:
```diff
@@ -13,15 +13,19 @@
-            {{ form_row(filterForm.minimalPrice, {label: 'Price from'|trans, symbolAfterInput: currencySymbolByDomainId(getDomain().id), attr: { class: 'js-product-filter-call-change-after-reset'} }) }}
-            {{ form_row(filterForm.maximalPrice, {label: 'Price to'|trans, symbolAfterInput: currencySymbolByDomainId(getDomain().id), attr: { class: 'js-product-filter-call-change-after-reset'} }) }}
+            <strong>
+            {{ 'Price'|trans }}
```
indenation

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/list.html.twig:
```diff
@@ -52,6 +52,7 @@
 
                 <div class="js-product-list-with-filter">
 
+                    
```
?

kulincz replied:
I don't know why it's here .. maybe some temporary todo.js solution. I tried to find across js files some connection and there is no any. I removed it and it seems that everything works fine.

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -11,6 +11,53 @@
+        display: none;
+    }
+
+    .js-body-filters-open & {
```
if somebody change js class name, i mean programmer, he wouldnt know he removed overlay in mobile version.. thats why is better to you same "state" class instead of mixing js classes with css .. cos in ssfw are rules that js classes are only for js, css classes only for less.. it could be "web--with-mobile-overlay" or simply "overlay" on body

kulincz replied:
Ok, but I need some trigger that let my style these interaction. And how I should do it without using this js-class? There is no direct style for body (.js-body-filters-open) element ... I am using it only for triggering and styling other body child elements.

seedya replied:
i understand.. you can use js class for js function for adding class to body, and that function will append css class to body for ex. "active-overlay" for use in less (here).. just for prevent problem i wrote

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/disable.less:
```diff
@@ -1,3 +1,4 @@
     opacity: 0.5;
-}
\ No newline at end of file
+    cursor: not-allowed!important;
```
idk why you need to use important..

kulincz replied:
Because there is needed to have high specificity a on this component has only one purpose – when is aplied then I need to be sure there will be not-allowed cursor. How to do it without important?
![image](https://user-images.githubusercontent.com/29253585/44839268-a6220f80-ac3e-11e8-82ca-4c7ca1b4bc22.png)

seedya replied:
oh i understand :)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -58,7 +112,20 @@
+        @media @query-tablet {
+            .js-body-filters-open & {
+                position: fixed;
+                right: 10px;
```
maybe ist here also connection to web padding?

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig:
```diff
@@ -29,22 +29,22 @@
-                    {% endif %}
-                </span>
-            </label>
+            <div class="in-styled-checkbox__wrap">
```
yeah, here you have component element outside of component.. i did that before too, but thats not BEM :)

kulincz replied:
Ok, teach me please :) how to solve it better?

seedya replied:
what i do in this case is to change structure from in-styled-checkbox__wrap > in-styled-checkbox to in-styled-checkbox > in-styled-checkbox__in..  :))

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/flag.less:
```diff
@@ -7,6 +7,9 @@
     top: @in-flag-position-top;
     left: @in-flag-position-left;
+
+    color: @color-f;
```
i like to use new variable - @color-base-inverted ... its always white, but i like it more :)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/products.less:
```diff
@@ -167,22 +190,29 @@
             margin-bottom: @list-products-item-inner-gap;
+            width: 100%;
+
+
```
2 empty lines

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/layout/header/cart.less:
```diff
@@ -36,8 +36,12 @@
                 color: @color-f;
             }
 
+            .cart--full:hover & {
```
cart is never full :D

kulincz replied:
ok, cart--with-items will be better :)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -35,6 +35,8 @@
             overflow-y: auto;
 
             background-color: #ffffff;
```
use variables

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/filter.less:
```diff
@@ -35,6 +35,8 @@
 
             background-color: #ffffff;
+
+            -webkit-overflow-scrolling: touch;
```
interesting.. have you post it to coders team what was problem?

kulincz replied:
I don' understand what is interesting here? :-) It's needed solution for smooth scrolling on touch devices with webit ... it's well-know think I guess https://css-tricks.com/snippets/css/momentum-scrolling-on-ios-overflow-elements/

seedya replied:
interesting is that i heard about this for the first time :))

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/form/input-spinbox.less:
```diff
@@ -27,9 +35,23 @@
         color: @color-base;
 
+        @media @query-tablet {
+            width: @form-input-spinbox-buttons-width*1.5;
```
will your math work when somebody change px of @form-input-spinbox-buttons-width?

kulincz replied:
I tried it and it seems ok.

seedya replied:
nice

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/flag.less:
```diff
@@ -12,6 +12,11 @@
 
     .clearfix();
 
+    .box-detail__image & {
```
i dont like mixing of components .. you can set default position for product detail and in product list use modification to set different position..

kulincz replied:
Ok, understood. I tried to fix it but there in now flags in admin and frontend on demoshop any now. So I erased these not-nice blocks from flag.less .. I don't know if let it here or remove it completely from main.less .. what do you think?

seedya replied:
let it here, if there are any, its not problem :)

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig:
```diff
@@ -20,99 +20,101 @@
-                                    {{ image(cartItem.product, { size: 'thumbnail' }) }}
-                                </a>
-                            </td>
+            <div class="table-cart__wrapper">
```
component element outside of component.. better is to set div to table-cart and table call table-cart__in or table-cart__table

kulincz replied:
Ok, but there is a lot more code depends on table-cart class and all I need here is wrapper on this place ... there wasn't easy to change it .. this was cheaper solution.

seedya replied:
you dont need to add __in__ into inner elements.. so its only change 2 class names.. i think :D

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/btn.less:
```diff
@@ -43,54 +44,54 @@
+    @btn-primary-color: @color-f;
+    @btn-primary-color-hover: @color-f;
+    @btn-primary-bg: @color-blue;
+    @btn-primary-bg-hover: darken(@btn-primary-bg, 5%);
```
is it better to write it to the top?

kulincz replied:
Sure ... this is not my job :-) moved to top ..

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/order.less:
```diff
@@ -23,6 +24,18 @@
             }
         }
+
+        fieldset {
```
add class to theese elements

kulincz replied:
ok, fixed.

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/order.less:
```diff
@@ -42,4 +55,24 @@
+            padding: 20px 20px 0 20px;
+        }
+
+        &--all-padding {
```
when i look at the name of modification i dont know what it means.. what about something like --same-padding?

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/products.less:
```diff
@@ -258,19 +258,25 @@
                 }
             }
+
+            .btn--success {
```
better to use &__button .. then fix it to the name of modification

kulincz replied:
It's because I don't know how to add class to this:
![image](https://user-images.githubusercontent.com/29253585/44842142-60b51080-ac45-11e8-8441-9f26692bf2f1.png)

kulincz replied:
temporary I did this:
![image](https://user-images.githubusercontent.com/29253585/44842182-7cb8b200-ac45-11e8-990f-0fa47ed93e92.png)

seedya replied:
oh yeah, here its hard to do it, for demo it doesnt worth it

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/table/cart-preview.less:
```diff
@@ -25,5 +31,13 @@
             }
         }
+
+        &--all-price-inc-vat {
```
include.. shortcuts ruines readability

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Inline/Paginator/paginator.html.twig:
```diff
@@ -18,7 +18,9 @@
                     {% set routeParams = routeParams|merge({'page': page}) %}
-                    <a class="in-paging__control__item in-paging__control__item--arrow" href="{{ path(route, routeParams) }}" rel ="prev">&lt;</a>
+                    <a class="in-paging__control__item in-paging__control__item--arrow" href="{{ path(route, routeParams) }}" rel ="prev">
+                        <span class="svg svg-arrow in-rotate--90"></span>
```
idk .. when i use modification i use component too.. here it looks like helper with name like modification

kulincz replied:
I don't understand .. where is the problem? :-)

seedya replied:
in BEM you have Block element and modifier .. everytime you use modifier, it belongs to some block.. but there is no block only modifier.. for me in-rotate--90 looks like helper, because it has only one attribute and block in-rotate is useless

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/layout/footer.less:
```diff
@@ -32,11 +44,17 @@
             flex: 0 @footer-newsletter-form-width;
         }
 
+        .form-choice__label {
```
combination of different component element.. mmm.. no.. use some modification or different class

kulincz replied:
Hmm, great tip .. but it's not so easy to modificate it .. if you know how to do it, let me know and I will fix it:
![image](https://user-images.githubusercontent.com/29253585/44842615-a2928680-ac46-11e8-8a6a-fda5f308ad6d.png)

seedya replied:
mmm okay leave it like that :D

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less:
```diff
@@ -48,6 +48,15 @@ img {
             background: @web-header-background;
         }
+
+        &--box-advantages {
```
is it necessary to call modification of layout line by its content especially with name of component? what about this solution - use --dark and add new one --with-bottom-border ? Or rename this to --dark-with-border .. i think its good to further usage and everyboy knows what this modification do

seedya commented on src/Shopsys/ShopBundle/Resources/views/Front/Layout/layout.html.twig:
```diff
@@ -35,6 +35,63 @@
         </div>
     </div>
+    {% block advantages %}
+    <div class="web__advantages">
```
i dont like to use in layout names, which are connected to its content (maybe only if its necessary)

kulincz replied:
This is block of standard e-commerce advantages in line ... what is better name for this kind of block? :-)

seedya replied:
i use name from what it do... line, container etc.. but now i see it has no css properties.. so i dont know purpose of this class.. but it doesnt matter :)

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/panel-callus.less:
```diff
@@ -0,0 +1,97 @@
+@box-panel-callus-icon-size: 20px;
+
+.box-panel-callus {
```
call-us

seedya commented on src/Shopsys/ShopBundle/Resources/styles/front/common/components/table/cart-preview.less:
```diff
@@ -1,6 +1,8 @@
 .table-cart-preview {
     margin-bottom: @margin-gap;
     width: 100%;
+    position: sticky;
```
woooow

kulincz replied:
what? :D

seedya replied:
i didnt know that exist.. perfect to know :D 👍

seedya reviewed (commented):
Hi, i have edit few notes, ples check them out

kulincz committed:
List menu variable fix

kulincz committed:
CSS stylistics

kulincz committed:
Search img size in variable

kulincz committed:
Indentation connect via variable

kulincz committed:
Styled checkbox moved to form components

kulincz committed:
Filters fixes

kulincz committed:
Variables fix

kulincz committed:
CSS lint

kulincz committed:
CSS fixing: filters, flag, btn, cart

kulincz committed:
Fieldset has class now

kulincz committed:
Naming

kulincz committed:
class naming

kulincz committed:
CSS naming

kulincz commented:
@seedya Please check my fixes and comments and let me know .. thank you for your CR! ;-)

seedya commented:
@kulincz i answered and checked changes, its okay! :)

vitek-rostislav committed:
.window-bar (info bar on FE) is now visible again

- it was hidden for design debugging purposes

vitek-rostislav committed:
3rd step of order: readonly inputs for delivery address are now greyed out
